### PR TITLE
Ability to request with a certain scope

### PR DESCRIPTION
**Pull request by rhaskia**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,10 +10,9 @@ repository = "https://github.com/jakewilkins/gh-device-flow"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-reqwest = { version = "0.11", features = ["blocking", "json"] }
+reqwest = { version = "0.11", features = ["blocking", "json", "rustls-tls"] }
```
Comment (jakewilkins) [inline Cargo.toml]: Can you provide some more context here on why this change is required and what sort of validation you've done that it is non-breaking?

**Review by jakewilkins — COMMENTED** [review]
Thanks for the PR!

I use this with GitHub Apps so passing scopes never crossed my mind 🤦‍♂️ 

I'd like to merge this but could use a little more detail on some of the changes.

File: Cargo.toml (hunk 1 above)
Comment (rhaskia) [reply Cargo.toml]: Openssl-sys was taking a long time to compile on my end, so I found it could be replaced with rust-ls which seems to run better. Reqwest hasn't thrown any errors in the time that I've switched so I don't think there is much difference apart from performance.

Comment (jakewilkins) [reply Cargo.toml]: `rustls-tls` fails to compile correctly on the `arm` or `armv7` builds.

If you know of a simple way to make those work I'm glad to hear it, or if you're okay with merging without the `rustls-tls` change I'm happy to have that as well.

Comment (rhaskia) [reply Cargo.toml]: If it still runs as normal without it that is good with me.

Comment (jakewilkins) [reply Cargo.toml]: I know that there is a way for me to make this work that would allow you to choose the one that's best for you but I haven't ever done that before and don't have the time to mess with figuring it right now 😅

Comment (rhaskia) [reply Cargo.toml]: What do you mean?

Comment (jakewilkins) [reply Cargo.toml]: What I had in mind was something similar to https://github.com/jakewilkins/gh-device-flow/pull/3 but I just kind of feel like I'm misusing Cargo in some way and if I could find the right options things would go better.

**Comment by rhaskia** [comment]
Oh, I wanted to upload an app which has this as a dependency to crates.io but I cannot as it seems the crates.io version of this does not have the scope feature yet. Would you be able to update the crates.io at some point when you are free?
